### DISCUSSION TITLE
How to get the borders of a mesh?

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#7170] [Q&A] [answered]
Hello,

With `PMP::extract_boundary_cycles`, we can get one halfedge per boundary cycle. But how to get all the halfedges forming one boundary cycle?

**Comment by afabri** [comment, marked as answer]
With [`halfedges_around_face()`](https://doc.cgal.org/latest/BGL/group__PkgBGLIterators.html#ga37229df31508a78eb4acdaf907e637e5)
